src/main/java/com/aws/greengrass/secretmanager/LocalStoreMap.java

```diff
@@ -0,0 +1,303 @@
+                //  used for encrypting and decrypting the secrets have changed.
+                try (LockScope ls = LockScope.lock(lock)) {
+                    crypter.set(null);
+                }
+            }
+        }));
+        this.secrets = new ConcurrentHashMap<>();
+    }
+
+    private boolean validUpdate(Node node, String key) {
+        return node != null && node.childOf(key) && Utils.isNotEmpty(Coerce.toString(node));
+    }
+
+    protected Crypter getCrypter() throws SecretCryptoException {
+        try (LockScope ls = LockScope.lock(lock)) {
+            if (crypter.get() == null) {
+                try {
+                    loadCrypter();
+                } catch (Exception e) {
+                    throw new SecretCryptoException("Unable to load crypter", e);
+                }
+            }
+            return crypter.get();
+        }
+    }
+
+    private void loadCrypter() throws Exception {
+        try {
+            URI privateKeyUri = securityService.getDeviceIdentityPrivateKeyURI();
+            URI certUri = securityService.getDeviceIdentityCertificateURI();
+            KeyPair kp = RetryUtils.runWithRetry(
+                    RetryUtils.RetryConfig.builder().maxRetryInterval(Duration.ofSeconds(30))
+                            .initialRetryInterval(Duration.ofSeconds(10)).maxAttempt(10)
+                            .retryableExceptions(Collections.singletonList(ServiceUnavailableException.class)).build(),
+                    () -> securityService.getKeyPair(privateKeyUri, certUri), "get-keypair", logger);
+            MasterKey masterKey = RSAMasterKey.createInstance(kp.getPublic(), kp.getPrivate());
+            KeyChain keyChain = new KeyChain();
+            keyChain.addMasterKey(masterKey);
+            crypter.set(new Crypter(keyChain));
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public void syncWithConfig(List<SecretConfiguration> secretConfiguration) {
+        reloadSecretsFromLocalStore();
+        save(secretConfiguration);
+    }
+
+    /**
+    Downloaded secrets are added to the local store based on the component configuration only.
+    Cache should be updated whenever the store is updated.
+     */
+    private void save(List<SecretConfiguration> secretConfiguration) {
+        List<AWSSecretResponse> responses = new ArrayList<>();
+        if (!secrets.isEmpty()) {
+            secretConfiguration.forEach((secretConfig) -> {
+                secretConfig.getLabels().forEach((label) -> {
+                    String arn = secretConfig.getArn();
+                    if (secrets.containsKey(arn) && secrets.get(arn).responseMap.containsKey(label)) {
+                        responses.add(secrets.get(arn).responseMap.get(label));
+                    }
+                });
+            });
+        }
+
+        try {
+            // update cache whenever the store is updated.
```

Comment:
If we want to update the cache when we update the store, should we be calling `reloadCache`?

Reply:
I'll move that comment over to the method description as it makes more sense there.